CodeGeneration/Sources/generate-swift-syntax/templates/swiftsyntax/RenamedChildrenCompatibilityFile.swift

```diff
@@ -16,90 +16,112 @@ import SyntaxSupport
-              collectionElementName: let collectionElementName,
-              defaultsToEmpty: _,
-              deprecatedCollectionElementName: let deprecatedCollectionElementName
-            ) = child.kind,
-            let deprecatedCollectionElementName
-          {
-            let childEltType = childNode.collectionElementType.syntaxBaseName
-
-            DeclSyntax(
-              """
-              @available(*, deprecated, renamed: "add\(raw: collectionElementName)")
-              public func add\(raw: deprecatedCollectionElementName)(_ element: \(childEltType)) -> \(layoutNode.kind.syntaxType) {
-                return add\(raw: collectionElementName)(element)
-              }
-              """
-            )
-          }
+      for child in deprecatedVars {
+        makeCompatibilityVar(for: child)
+        if let addMethod = makeCompatibilityAddMethod(for: child) {
+          addMethod
         }
       }
 
-      let deprecatedNames = layoutNode.children
-        .filter { !$0.isUnexpectedNodes && $0.hasDeprecatedName }
-        .map { $0.identifier.description }
-        .joined(separator: ", ")
-
-      let renamedArguments =
-        layoutNode.children.map { child in
-          if child.isUnexpectedNodes {
-            return "_:"
-          } else {
-            return "\(child.labelDeclName):"
-          }
-        }.joined(separator: "")
-
-      let renamedName = "\(layoutNode.type.syntaxBaseName)(leadingTrivia:\(renamedArguments)trailingTrivia:)"
-
-      try! InitializerDeclSyntax(
-        """
-        @available(*, deprecated, renamed: \(literal: renamedName))
-        @_disfavoredOverload
-        \(layoutNode.generateInitializerDeclHeader(useDeprecatedChildName: true))
-        """
-      ) {
-        FunctionCallExprSyntax(callee: ExprSyntax("self.init")) {
-          LabeledExprSyntax(label: "leadingTrivia", expression: ExprSyntax("leadingTrivia"))
-          for child in layoutNode.children {
-            if child.isUnexpectedNodes {
-              LabeledExprSyntax(expression: ExprSyntax("\(child.deprecatedVarName ?? child.baseCallName)"))
-            } else {
-              LabeledExprSyntax(
-                label: child.labelDeclName,
-                colon: .colonToken(),
-                expression: DeclReferenceExprSyntax(baseName: child.deprecatedVarName ?? child.baseCallName)
-              )
-            }
-          }
-          LabeledExprSyntax(label: "trailingTrivia", expression: ExprSyntax("trailingTrivia"))
+      let renamedName = InitSignature(layoutNode).compoundName
+      for signature in deprecatedInitSignatures {
+        makeCompatibilityInit(for: signature, renamedName: renamedName)
+      }
+    }
+  }
+
+  for trait in TRAITS.filter({ !$0.childHistory.isEmpty }) {
+    var deprecatedVars = SYNTAX_COMPATIBILITY_LAYERS.deprecatedVarsByTrait[trait.traitName] ?? []
+    var deprecatedArgLists = SYNTAX_COMPATIBILITY_LAYERS.deprecatedInitSignaturesByTrait[trait.traitName] ?? []
```

Comment:
I assume this raises an unused variable warning, right? Could you remove `deprecatedArgLists` to fix that?

Reply:
This might again be a change that ought to be deferred into a commit in the other PR—I'll take a look at it.

Reply:
Ended up tweaking this so there's a small struct grouping the vars and inits together; I'd been thinking about making that change for a while anyway.